api/lib/opentelemetry/trace/propagation/trace_context/text_map_propagator.rb

```diff
@@ -45,8 +45,12 @@ def inject(carrier, context: Context.current, setter: Context::Propagation.text_
           # @return [Context] context updated with extracted baggage, or the original context
           #   if extraction fails
           def extract(carrier, context: Context.current, getter: Context::Propagation.text_map_getter)
-            tp = TraceParent.from_string(getter.get(carrier, TRACEPARENT_KEY))
-            tracestate = Tracestate.from_string(getter.get(carrier, TRACESTATE_KEY))
+            trace_parent_value = getter.get(carrier, TRACEPARENT_KEY)
+            trace_state_value = getter.get(carrier, TRACESTATE_KEY)
+            return context if trace_parent_value.nil? || trace_state_value.nil?
```

Comment:
The [W3C Trace Context](https://www.w3.org/TR/trace-context/#tracestate-header) states that:
> If the vendor failed to parse `traceparent`, it _MUST NOT_ attempt to parse `tracestate`. Note that the opposite is not true: failure to parse `tracestate` _MUST NOT_ affect the parsing of `traceparent`.

so we must not return early if `trace_state` is `nil`.
```suggestion
            trace_parent_value = getter.get(carrier, TRACEPARENT_KEY)
            trace_state_value = getter.get(carrier, TRACESTATE_KEY)
            return context if trace_parent_value.nil?
```